### ISSUE TITLE
PublicApiAnalyzers doesn't track delegate parameter types or return type

### ISSUE THREAD

**Issue by carlreinke** [dotnet/roslyn-analyzers#6003]
### Analyzer

**Diagnostic ID**: [RS0016](https://github.com/dotnet/roslyn-analyzers/blob/main/src/PublicApiAnalyzers/Microsoft.CodeAnalysis.PublicApiAnalyzers.md): `Add public types and members to the declared API`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.PublicApiAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.PublicApiAnalyzers/)

**Version**: 3.3.3 (Latest)

### Steps To Reproduce

1. Code up a public delegate:
    ```
    namespace PublicApiAnalyzerTest
    {
        public delegate int Test();
    }
    ```
1. Use the fixer to add the API to `PublicAPI.Unshipped.txt`.  The added line is ```PublicApiAnalyzerTest.Test```.
1. Change the return type of the delgate.

### Expected behavior

There should be a diagnostic since a public API changed.

### Actual behavior

A public API changed, but no diagnostic is reported.
